src/_pytest/stepwise.py

```diff
@@ -0,0 +1,104 @@
+from _pytest.cacheprovider import Cache
+import pytest
+
+
+def pytest_addoption(parser):
+    group = parser.getgroup("general")
+    group.addoption(
+        "--sw",
+        "--stepwise",
+        action="store_true",
+        dest="stepwise",
+        help="exit on test fail and continue from last failing test next time",
+    )
+    group.addoption(
+        "--stepwise-skip",
+        action="store_true",
+        dest="stepwise_skip",
+        help="ignore the first failing test but stop on the next failing test",
+    )
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_configure(config):
+    config.cache = Cache.for_config(config)
```

Comment:
We need to remove this line because `cacheprovider` already creates the cache for us:

https://github.com/pytest-dev/pytest/blob/f7285b6ab2222e29d80ad1501a301c250db444e0/src/_pytest/cacheprovider.py#L294-L297

We should also remove the `tryfirst=True` option from the hook implementation so `cacheprovider` executes first.